src/ds/bits.h

```diff
@@ -1,6 +1,7 @@
 #pragma once
 
 #include <limits>
+#include <cstddef>
```

Comment:
What is this needed for?

Reply:
size_t definition

Reply:
Why don't we hit that on non-ARM?

Reply:
arm esp 32 bits is pretty limited arch in comparison. header arranged differently and all. that was coming from compiler build warning would not have caught this otherwise.